diff --git a/debian/control b/debian/control
index fb3d81f2..a8f07273 100644
--- a/debian/control
+++ b/debian/control
@@ -57,6 +57,8 @@ Depends: plasma-framework,
          qml-module-org-kde-kopeninghours,
          qml-module-org-kde-kosmindoormap,
          qml-module-org-kde-kpublictransport,
+         qml-module-org-kde-kirigami-addons-datetime,
+         qml-module-org-kde-kirigami-addons-labs-mobileform,
          qml-module-org-kde-prison,
          qml-module-qt-labs-platform,
          qml-module-qt-labs-qmlmodels,
