From 1ed26e451022e941627c5e1eb499eb876e3ac50e Mon Sep 17 00:00:00 2001
From: Petter Reinholdtsen <pere@debian.org>
Date: Tue, 21 Feb 2023 04:14:07 +0100
Subject: [PATCH] Fixed typo in d//source/lintian-overrides.

---
 debian/source/lintian-overrides | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/source/lintian-overrides b/debian/source/lintian-overrides
index 93f5703f..30d220df 100644
--- a/debian/source/lintian-overrides
+++ b/debian/source/lintian-overrides
@@ -1,4 +1,4 @@
 # The full commented source is there.
-source-is-missing [src/libclient/web-chatview/linkify.j]
+source-is-missing [src/libclient/web-chatview/linkify.js]
 source-is-missing [src/libclient/web-chatview/jed.js]
 source-is-missing [3rdparty/SortFilterProxyModel/docs/*.html]
-- 
GitLab

