--- src/akregator_part.cpp	2020-06-05 11:41:23.000000000 -0500
+++ src/akregator_part_trisquel.cpp	2020-06-05 12:46:05.914051386 -0500
@@ -189,6 +189,66 @@
     wire.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("http://wire.kubuntu.org/?feed=rss2"));
     kubuntuFolder.appendChild(wire);
 
+// Trisquel feeds
+    QDomElement trisquelFolder = doc.createElement(QStringLiteral("outline"));
+    trisquelFolder.setAttribute(QStringLiteral("text"), QStringLiteral("Trisquel GNU/Linux"));
+    body.appendChild(trisquelFolder);
+
+    QDomElement tnews = doc.createElement(QStringLiteral("outline"));
+    tnews.setAttribute(QStringLiteral("text"), i18n("News"));
+    tnews.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://trisquel.info/en/node/feed"));
+    trisquelFolder.appendChild(tnews);
+
+    QDomElement relAnnounce = doc.createElement(QStringLiteral("outline"));
+    relAnnounce.setAttribute(QStringLiteral("text"), i18n("Release announcements"));
+    relAnnounce.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://trisquel.info/en/taxonomy/term/700/0/feed"));
+    trisquelFolder.appendChild(relAnnounce);
+
+    QDomElement enForum = doc.createElement(QStringLiteral("outline"));
+    enForum.setAttribute(QStringLiteral("text"), i18n("Forum"));
+    enForum.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://trisquel.info/en/taxonomy/term/50/0/feed"));
+    trisquelFolder.appendChild(enForum);
+
+    QDomElement tReddit = doc.createElement(QStringLiteral("outline"));
+    tReddit.setAttribute(QStringLiteral("text"), i18n("Trisquel at Reddit"));
+    tReddit.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://www.reddit.com/r/trisquel/.rss"));
+    trisquelFolder.appendChild(tReddit);
+
+// Free Software feeds
+    QDomElement fsFolder = doc.createElement(QStringLiteral("outline"));
+    fsFolder.setAttribute(QStringLiteral("text"), QStringLiteral("Free Software"));
+    body.appendChild(fsFolder);
+
+    QDomElement newGNU = doc.createElement(QStringLiteral("outline"));
+    newGNU.setAttribute(QStringLiteral("text"), i18n("What's New at GNU"));
+    newGNU.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://www.reddit.com/r/trisquel/.rss"));
+    fsFolder.appendChild(newGNU);
+
+    QDomElement fsfNews = doc.createElement(QStringLiteral("outline"));
+    fsfNews.setAttribute(QStringLiteral("text"), i18n("FSF News"));
+    fsfNews.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://static.fsf.org/fsforg/rss/news.xml"));
+    fsFolder.appendChild(fsfNews);
+
+    QDomElement fsfEvents = doc.createElement(QStringLiteral("outline"));
+    fsfEvents.setAttribute(QStringLiteral("text"), i18n("FSF Events"));
+    fsfEvents.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://static.fsf.org/fsforg/rss/events.xml"));
+    fsFolder.appendChild(fsfEvents);
+
+    QDomElement fsfBlog = doc.createElement(QStringLiteral("outline"));
+    fsfBlog.setAttribute(QStringLiteral("text"), i18n("FSF Blogs"));
+    fsfBlog.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://static.fsf.org/fsforg/rss/blogs.xml"));
+    fsFolder.appendChild(fsfBlog);
+
+    QDomElement fsDaily = doc.createElement(QStringLiteral("outline"));
+    fsDaily.setAttribute(QStringLiteral("text"), i18n("fsdaily.com - Free Software News"));
+    fsDaily.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://www.fsdaily.com/feed/published/All"));
+    fsFolder.appendChild(fsDaily);
+
+    QDomElement gnuReddit = doc.createElement(QStringLiteral("outline"));
+    gnuReddit.setAttribute(QStringLiteral("text"), i18n("GNU at Reddit"));
+    gnuReddit.setAttribute(QStringLiteral("xmlUrl"), QStringLiteral("https://www.reddit.com/r/gnu/.rss"));
+    fsFolder.appendChild(gnuReddit);
+
     return doc;
 }
 }
