diff --git a/debian/rules b/debian/rules
index 7b8687f1..c5023465 100755
--- a/debian/rules
+++ b/debian/rules
@@ -12,7 +12,7 @@ override_dh_auto_clean:
 	rm -rf debian/build
 
 override_dh_auto_configure:
-	dh_auto_configure -- -Dclapper=true
+	dh_auto_configure -- -Dclapper=false
 
 override_dh_auto_test:
 ifeq (,$(filter nocheck,$(DEB_BUILD_OPTIONS)))
diff --git a/debian/control b/debian/control
index 6f7b4072..11ffa147 100644
--- a/debian/control
+++ b/debian/control
@@ -7,7 +7,7 @@ Uploaders: Federico Ceratto <federico@debian.org>,
 Rules-Requires-Root: no
 Build-Depends:
  appstream-util,
- clapper (>= 0.6.0~),
+ clapper (>= 0.5.2~),
  debhelper-compat (= 13),
  desktop-file-utils,
  gettext,
