diff --git a/applets/wncklet/org.mate.panel.applet.workspace-switcher.gschema.xml.in b/applets/wncklet/org.mate.panel.applet.workspace-switcher.gschema.xml.in
index c49f472d..c9c88998 100644
--- a/applets/wncklet/org.mate.panel.applet.workspace-switcher.gschema.xml.in
+++ b/applets/wncklet/org.mate.panel.applet.workspace-switcher.gschema.xml.in
@@ -11,7 +11,7 @@
       <description>If true, the workspace switcher will show all workspaces. Otherwise it will only show the current workspace.</description>
     </key>
     <key name="num-rows" type="i">
-      <default>1</default>
+      <default>2</default>
       <summary>Rows in workspace switcher</summary>
       <description>This key specifies how many rows (for horizontal layout) or columns (for vertical layout) the workspace switcher shows the workspaces in. This key is only relevant if the display_all_workspaces key is true.</description>
     </key>
