diff -Nru baobab/data/Makefile.am baobab/data/Makefile.am
--- a/baobab/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/baobab/data/Makefile.am	2022-02-16 16:41:58.357104436 -0600
@@ -6,7 +6,7 @@
 	sed -e "s|\@VERSION\@|@VERSION@|" $< > $@
 
 $(baobabapp_DATA): $(baobabapp_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 appdatadir = $(datadir)/metainfo
 appdata_in_files = mate-disk-usage-analyzer.appdata.xml.in
diff -Nru gsearchtool/data/Makefile.am gsearchtool/data/Makefile.am
--- a/gsearchtool/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/gsearchtool/data/Makefile.am	2022-02-16 16:41:59.341116566 -0600
@@ -18,7 +18,7 @@
 Utilities_in_files = mate-search-tool.desktop.in
 Utilities_DATA = $(Utilities_in_files:.desktop.in=.desktop)
 $(Utilities_DATA): $(Utilities_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 EXTRA_DIST = \
 	$(icon_DATA) \
diff -Nru logview/data/Makefile.am logview/data/Makefile.am
--- a/logview/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/logview/data/Makefile.am	2022-02-16 16:41:59.989124555 -0600
@@ -6,7 +6,7 @@
 $(desktop_in_files): $(desktop_in_files:.desktop.in=.desktop.in.in)
 	@sed -e "s|\@VERSION\@|@VERSION@|" $< > $@
 $(desktop_DATA): $(desktop_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 man_MANS = mate-system-log.1
 
diff -Nru mate-dictionary/data/Makefile.am mate-dictionary/data/Makefile.am
--- a/mate-dictionary/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/mate-dictionary/data/Makefile.am	2022-02-16 16:42:00.853135206 -0600
@@ -9,6 +9,9 @@
 $(gdictapp_in_files): $(gdictapp_in_files:.desktop.in=.desktop.in.in)
 	$(AM_V_GEN)sed -e "s|\@VERSION\@|@VERSION@|" $< > $@
 
+$(gdictapp_DATA): $(gdictapp_in_files)
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
+
 # dictionary sources definitions
 dictsourcedir = $(datadir)/mate-dict/sources
 dictsource_in_files = \
@@ -17,8 +20,8 @@
 	$(NULL)
 dictsource_DATA = $(dictsource_in_files:.desktop.in=.desktop)
 
-%.desktop: %.desktop.in
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+$(dictsource_DATA): $(dictsource_in_files)
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Description --template $< -d $(top_srcdir)/po -o $@
 
 builderdir = $(datadir)/mate-dictionary
 builder_DATA = \
diff -Nru mate-disk-image-mounter/data/Makefile.am mate-disk-image-mounter/data/Makefile.am
--- a/mate-disk-image-mounter/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/mate-disk-image-mounter/data/Makefile.am	2022-02-16 16:42:02.137151035 -0600
@@ -4,7 +4,7 @@
 diskimagemounter_in_files = mate-disk-image-mounter.desktop.in
 diskimagemounter_DATA = $(diskimagemounter_in_files:.desktop.in=.desktop)
 $(diskimagemounter_DATA): $(diskimagemounter_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 EXTRA_DIST = 					\
 	$(diskimagemounter_in_files)		\
diff -Nru mate-screenshot/data/Makefile.am mate-screenshot/data/Makefile.am
--- a/mate-screenshot/data/Makefile.am	2020-02-09 19:39:11.000000000 -0600
+++ b/mate-screenshot/data/Makefile.am	2022-02-16 16:42:02.157151282 -0600
@@ -2,7 +2,7 @@
 mate_screenshot_in_files = mate-screenshot.desktop.in
 mate_screenshot_DATA = $(mate_screenshot_in_files:.desktop.in=.desktop)
 $(mate_screenshot_DATA): $(mate_screenshot_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 man_MANS = mate-screenshot.1
 
diff -Nru po/Makevars po/Makevars
--- a/po/Makevars	2020-02-09 19:39:11.000000000 -0600
+++ b/po/Makevars	2022-02-16 16:42:02.233152218 -0600
@@ -8,7 +8,7 @@
 top_builddir = ..
 
 # These options get passed to xgettext.
-XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --add-comments=Translators:
+XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --keyword=Description --add-comments=Translators:
 
 # This is the copyright holder that gets inserted into the header of the
 # $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
