diff --git a/debian/scripts/control-create b/debian/scripts/control-create
index 887f2863..faacf67f 100644
--- a/debian/scripts/control-create
+++ b/debian/scripts/control-create
@@ -70,23 +70,12 @@ gen_per_flavour () {
 		-e "s/TARGET/${target}/g"
 	)
 
-	if [ "${#signed_arch[@]}" != 0 ]; then
-		sed "${sed_common_patterns[@]}" \
-		    -e "s/ARCH/${signed_arch[*]}/g" \
-		    -e "s/=SIGN-ME-PKG=/-unsigned/g" \
-		    -e "s/=SIGN-ME-TXT=/ unsigned/g" \
-		    -e "s/=SIGN-PEER-PKG=//g" \
-		  "${DEBIAN}/control.d/flavour-signed-control.stub"
-	fi
-
-	if [ "${#unsigned_arch[@]}" != 0 ]; then
-		sed "${sed_common_patterns[@]}" \
-		    -e "s/ARCH/${unsigned_arch[*]}/g" \
-		    -e "s/=SIGN-ME-PKG=//g" \
-		    -e "s/=SIGN-ME-TXT=//g" \
-		    -e "s/=SIGN-PEER-PKG=/-unsigned/g" \
-		    "${DEBIAN}/control.d/flavour-signed-control.stub"
-	fi
+	sed "${sed_common_patterns[@]}" \
+	    -e "s/ARCH/${arch}/g" \
+	    -e "s/=SIGN-ME-PKG=//g" \
+	    -e "s/=SIGN-ME-TXT=//g" \
+	    -e "s/=SIGN-PEER-PKG=//g" \
+	    "${DEBIAN}/control.d/flavour-signed-control.stub"
 
 	sed "${sed_common_patterns[@]}" \
 	    -e "s/ARCH/${arch}/g" \
