diff -ru gnome-menus-3.2.0.orig/layout/applications.menu gnome-menus-3.2.0/layout/applications.menu
--- gnome-menus-3.2.0.orig/layout/applications.menu	2012-03-09 23:09:57.000000000 +0000
+++ gnome-menus-3.2.0/layout/applications.menu	2012-03-09 23:20:35.000000000 +0000
@@ -236,7 +236,7 @@
     </Include>
   </Menu> <!-- End Office -->
 
-  <!-- System Tools-->
+  <!-- System Tools is disabled in Trisquel, goes into gnome-control-center
   <Menu>
     <Name>System</Name>
     <Directory>System-Tools.directory</Directory>
@@ -247,7 +247,7 @@
 	<Not><Category>Game</Category></Not>
       </And>
     </Include>
-  </Menu>   <!-- End System Tools -->
+  </Menu>   End System Tools -->
 
   <!-- Other -->
   <Menu>
@@ -257,7 +257,7 @@
     <Include>
       <And>
         <Not><Category>Core</Category></Not>
-        <Not><Category>Screensaver</Category></Not>
+        <Not><Category>Screensaver</Category></Not> <Not><Category>System</Category></Not> <Not><Category>Settings</Category></Not>
         <Not><Category>X-GNOME-Settings-Panel</Category></Not>
       </And>
     </Include>
