--- a/debian/mtpaint.desktop	2017-08-31 10:16:04.887637107 -0400
+++ b/debian/mtpaint.desktop	2017-08-31 10:16:13.410891946 -0400
@@ -6,7 +6,7 @@
 Comment[es_ES]=programa de pintura para crear pixel art y manipular fotos digitales
 Comment[pt_BR]=programa de pintura para criar pixel art e manipular fotos digitais
 Type=Application
-Exec=mtpaint %U
+Exec=mtpaint %f
 Icon=mtpaint
 Categories=Graphics;RasterGraphics;
 Keywords=graphics;editor
