--- src/video/fbcon/SDL_fbvideo.c.orig	2011-09-08 17:30:03.670583685 +0200
+++ src/video/fbcon/SDL_fbvideo.c	2011-09-08 17:30:24.230582737 +0200
@@ -46,7 +46,6 @@
 #include "SDL_fbevents_c.h"
 #include "SDL_fb3dfx.h"
 #include "SDL_fbmatrox.h"
-#include "SDL_fbriva.h"
 
 /*#define FBCON_DEBUG*/
 
@@ -769,13 +768,6 @@
 #endif
 			FB_3DfxAccel(this, finfo.accel);
 			break;
-		    case FB_ACCEL_NV3:
-		    case FB_ACCEL_NV4:
-#ifdef FBACCEL_DEBUG
-			printf("NVidia hardware accelerator!\n");
-#endif
-			FB_RivaAccel(this, finfo.accel);
-			break;
 		    default:
 #ifdef FBACCEL_DEBUG
 			printf("Unknown hardware accelerator.\n");
