From 82e38d9e1c06e7fdc0816fe08f4021b8c1cdff10 Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Wed, 20 Jul 2022 11:25:26 +0900
Subject: [PATCH] engine: Delete xkb:latam:*:* in denylist.txt

BUG=https://github.com/ibus/ibus/issues/2412

diff --git a/engine/Makefile.am b/engine/Makefile.am
index 03867f52f..7256fbc82 100644
--- a/engine/Makefile.am
+++ b/engine/Makefile.am
@@ -4,7 +4,7 @@
 #
 # Copyright (c) 2010-2016, Google Inc. All rights reserved.
 # Copyright (c) 2007-2016 Peng Huang <shawn.p.huang@gmail.com>
-# Copyright (c) 2013-2020 Takao Fujiwara <takao.fujiwara1@gmail.com>
+# Copyright (c) 2013-2021 Takao Fujiwara <takao.fujiwara1@gmail.com>
 #
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of the GNU Lesser General Public
@@ -88,6 +88,7 @@ CLEANFILES = \
 	$(NULL)
 
 EXTRA_DIST = \
+	denylist.txt \
 	gensimple.py \
 	iso639converter.py \
 	simple.xml.in \
