From 8de87da89372d778f490aee8a9198c22521c97d2 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sun, 31 Oct 2021 18:04:36 +0100
Subject: [PATCH] Keyboard/kbdcompiler: Avoid 'custom' layout

Closes: #997110.
---
 Keyboard/kbdcompiler | 1 +
 debian/changelog     | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/Keyboard/kbdcompiler b/Keyboard/kbdcompiler
index 61c96af..726b23d 100755
--- a/Keyboard/kbdcompiler
+++ b/Keyboard/kbdcompiler
@@ -93,6 +93,7 @@ printf STDERR "Compiling compact keymaps for %s...\n", $model;
 for my $layout (values %KeyboardNames::layouts) {
     next if ($layout eq 'nec/jp');
     next if ($layout eq 'nec_vndr/jp');
+    next if ($layout eq 'custom');
     my $actual_layout = (($layout eq 'rs') ? 'rs,rs'
 			 : (($layout eq 'lt') ? 'lt,lt'
 			    : (($nonlatin =~ / $layout /)
