diff -ru linux-2.6.35.orig/drivers/scsi/megaraid.c linux-2.6.35/drivers/scsi/megaraid.c
--- linux-2.6.35.orig/drivers/scsi/megaraid.c	2010-08-01 22:11:14.000000000 +0000
+++ linux-2.6.35/drivers/scsi/megaraid.c	2011-05-02 20:30:39.000000000 +0000
@@ -4995,9 +4995,14 @@
 		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
 	{PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_AMI_MEGARAID3,
 		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
+        {PCI_VENDOR_ID_AMI, PCI_DEVICE_ID_AMI_MEGARAID3,
+                HP_SUBSYS_VID, HP_NETRAID1M_SUBSYS_DID, 0, 0, 0},
+        {PCI_VENDOR_ID_AMI, PCI_DEVICE_ID_AMI_MEGARAID3,
+                HP_SUBSYS_VID, HP_NETRAID2M_SUBSYS_DID, 0, 0, 0},
 	{0,}
 };
 MODULE_DEVICE_TABLE(pci, megaraid_pci_tbl);
+#define HBA_SIGNATURE 0x3344
 
 static struct pci_driver megaraid_pci_driver = {
 	.name		= "megaraid_legacy",
diff -ru linux-2.6.35.orig/drivers/scsi/megaraid.h linux-2.6.35/drivers/scsi/megaraid.h
--- linux-2.6.35.orig/drivers/scsi/megaraid.h	2010-08-01 22:11:14.000000000 +0000
+++ linux-2.6.35/drivers/scsi/megaraid.h	2011-05-02 20:31:01.000000000 +0000
@@ -84,6 +84,9 @@
 #define LSI_SUBSYS_VID			0x1000
 #define INTEL_SUBSYS_VID		0x8086
 
+#define HP_NETRAID1M_SUBSYS_DID 0x60E7
+#define HP_NETRAID2M_SUBSYS_DID 0x60E8
+
 #define HBA_SIGNATURE	      		0x3344
 #define HBA_SIGNATURE_471	  	0xCCCC
 #define HBA_SIGNATURE_64BIT		0x0299
