diff --git a/debian/netcfg-common.templates b/debian/netcfg-common.templates
index 9e3fae52..9d8c27b3 100644
--- a/debian/netcfg-common.templates
+++ b/debian/netcfg-common.templates
@@ -144,7 +144,7 @@ _Description: Failure of key exchange and association
 
 Template: netcfg/get_hostname
 Type: string
-Default: debian
+Default: trisquel
 # :sl1:
 _Description: Hostname:
  Please enter the hostname for this system.
