diff --git a/build/config/x86.cfg b/build/config/x86.cfg
index 5ddc8cef..74ddad61 100644
--- a/build/config/x86.cfg
+++ b/build/config/x86.cfg
@@ -443,8 +443,8 @@ arch_netboot_dir: x86_syslinux x86_grub_efi
 	mkdir -p $(TEMP_DEPTHCHARGE)
 	depthchargectl build -v \
 		--board amd64-generic \
-		--kernel-release $(KERNELVERSION) \
+		--kernel-release '' \
 		--kernel $(TEMP_KERNEL) \
 		--initramfs $(TEMP_INITRD) \
 		--root none \

