--- a/debian/control    2023-04-06 12:43:17.147156435 -0600
+++ b/debian/control    2023-04-06 12:46:32.883071182 -0600
@@ -34,11 +34,9 @@
          ubuntu-release-upgrader-gtk,
          gnome-shell <!s390x> | notification-daemon <!s390x>,
          policykit-1
-Recommends: apport-gtk (>=2.8-0ubuntu3), 
-            python3-aptdaemon.gtk3widgets | synaptic (>= 0.75.12), 
+Recommends: python3-aptdaemon.gtk3widgets | synaptic (>= 0.75.12), 
             software-properties-gtk, 
-            python3-aptdaemon,
-            whoopsie (>= 0.2.77)
+            python3-aptdaemon
 Description: Daemon which notifies about package updates
  Puts an icon in the user's notification area when package updates are
  available.
