diff --git a/data/os/trisquel.info/trisquel-10.xml.in b/data/os/trisquel.info/trisquel-10.xml.in
index 78fd5bdd..bc95ba98 100644
--- a/data/os/trisquel.info/trisquel-10.xml.in
+++ b/data/os/trisquel.info/trisquel-10.xml.in
@@ -12,7 +12,7 @@
     <distro>trisquel</distro>
     <codename>Nabia</codename>
     <upgrades id="http://trisquel.info/trisquel/9"/>
-    <derives-from id="http://ubuntu.com/ubuntu/20.04"/>
+    <derives-from id="http://trisquel.info/trisquel/9"/>
 
     <release-date>2022-02-01</release-date>
 
diff --git a/data/os/trisquel.info/trisquel-11.xml.in b/data/os/trisquel.info/trisquel-11.xml.in
index 8619adb1..294963f6 100644
--- a/data/os/trisquel.info/trisquel-11.xml.in
+++ b/data/os/trisquel.info/trisquel-11.xml.in
@@ -12,7 +12,7 @@
     <distro>trisquel</distro>
     <codename>Aramo</codename>
     <upgrades id="http://trisquel.info/trisquel/10"/>
-    <derives-from id="http://ubuntu.com/ubuntu/22.04"/>
+    <derives-from id="http://trisquel.info/trisquel/10"/>
 
     <release-date>2023-03-19</release-date>
 
