From a61542ffc2d3807dbc3163d1727cc5d8c2118838 Mon Sep 17 00:00:00 2001
From: mbkma <johannes.unruh@fau.de>
Date: Wed, 11 Jan 2023 20:53:29 +0100
Subject: [PATCH] rename Europe/Kiev to Europe/Kyiv

---
 data/Locations.xml.in | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/data/Locations.xml.in b/data/Locations.xml.in
index 8bf9bad..ea75673 100644
--- a/data/Locations.xml.in
+++ b/data/Locations.xml.in
@@ -20627,13 +20627,12 @@
       <iso-code>UA</iso-code>
       <fips-code>UP</fips-code>
       <timezones>
-        <timezone id="Europe/Kiev">
-          <obsoletes>Europe/Simferopol</obsoletes>
+        <timezone id="Europe/Kyiv">
           <obsoletes>Europe/Uzhgorod</obsoletes>
           <obsoletes>Europe/Zaporozhye</obsoletes>
         </timezone>
       </timezones>
-      <tz-hint>Europe/Kiev</tz-hint>
+      <tz-hint>Europe/Kyiv</tz-hint>
       <city>
         <!-- A city in Ukraine -->
         <name>Boryspil'</name>
@@ -20706,10 +20705,10 @@
       </city>
       <city>
         <!-- The capital of Ukraine.
-             "Kiev" is the traditional English name.
+             "Kyiv" is the traditional English name.
              The local name in Ukrainian is "Kyyiv".
           -->
-        <name>Kiev</name>
+        <name>Kyiv</name>
         <coordinates>50.433333 30.516667</coordinates>
         <location>
           <name>Kyiv</name>
