--- source/distro-info-util.c	2021-11-04 15:27:37.000000000 -0600
+++ source/distro-info-util.c_fix	2022-04-06 10:00:23.394992023 -0500
@@ -423,10 +423,6 @@
         free(list->distro->milestones[MILESTONE_EOL_LTS]);
         free(list->distro->milestones[MILESTONE_EOL_ELTS]);
 #endif
-#ifdef UBUNTU
-        free(list->distro->milestones[MILESTONE_EOL_SERVER]);
-        free(list->distro->milestones[MILESTONE_EOL_ESM]);
-#endif
         free(list->distro);
         free(list);
         list = next;
@@ -581,7 +577,7 @@
            "  -t  --testing          current testing version\n"
 #endif
            "  -s  --stable           latest stable version\n"
-#ifdef UBUNTU
+#ifdef TRISQUEL
            "      --lts              latest long term support (LTS) version\n"
 #endif
 #ifdef DEBIAN
@@ -592,9 +588,6 @@
            "  -l  --lts              list of all LTS supported versions\n"
            "  -e  --elts             list of all Extended LTS supported versions\n"
 #endif
-#ifdef UBUNTU
-           "      --supported-esm    list of all Ubuntu Advantage supported stable versions\n"
-#endif
            "      --unsupported      list of all unsupported stable versions\n"
            "  -c  --codename         print the codename (default)\n"
            "  -f  --fullname         print the full name\n"
@@ -612,7 +605,7 @@
 #ifdef DEBIAN
             "--elts, "
 #endif
-#ifdef UBUNTU
+#ifdef TRISQUEL
             "--latest, "
 #endif
             "--lts, "
@@ -620,9 +613,6 @@
             "--oldstable, "
 #endif
             "--stable, --supported, "
-#ifdef UBUNTU
-            "--supported-esm, "
-#endif
             "--series, "
 #ifdef DEBIAN
             "--testing, "
@@ -651,7 +641,7 @@
 #ifdef DEBIAN
     char *alias_codename = NULL;
 #endif
-#ifdef UBUNTU
+#ifdef TRISQUEL
     bool filter_latest = false;
 #endif
 
@@ -664,9 +654,6 @@
         {"devel",         no_argument,       NULL, 'd' },
         {"stable",        no_argument,       NULL, 's' },
         {"supported",     no_argument,       NULL, 'S' },
-#ifdef UBUNTU
-        {"supported-esm", no_argument,       NULL, 'e' },
-#endif
         {"unsupported",   no_argument,       NULL, 'U' },
         {"codename",      no_argument,       NULL, 'c' },
         {"fullname",      no_argument,       NULL, 'f' },
@@ -678,14 +665,14 @@
         {"oldstable",     no_argument,       NULL, 'o' },
         {"testing",       no_argument,       NULL, 't' },
 #endif
-#ifdef UBUNTU
+#ifdef TRISQUEL
         {"latest",        no_argument,       NULL, 'l' },
         {"lts",           no_argument,       NULL, 'L' },
 #endif
         {NULL,            0,                 NULL, '\0' }
     };
 
-#ifdef UBUNTU
+#ifdef TRISQUEL
     const char *short_options = "hadscrfly::";
 #endif
 #ifdef DEBIAN
@@ -731,7 +718,7 @@
             case 'd':
                 selected_filters++;
                 filter_cb = filter_devel;
-#ifdef UBUNTU
+#ifdef TRISQUEL
                 select_cb = select_latest_created;
 #endif
 #ifdef DEBIAN
@@ -782,7 +769,7 @@
                 free(date);
                 return EXIT_SUCCESS;
 
-#ifdef UBUNTU
+#ifdef TRISQUEL
             case 'l':
                 selected_filters++;
                 filter_latest = true;
@@ -836,15 +823,6 @@
                 select_cb = NULL;
                 break;
 
-#ifdef UBUNTU
-            case 'e':
-                // Only long option --supported-esm is used
-                selected_filters++;
-                filter_cb = filter_esm_supported;
-                select_cb = NULL;
-                break;
-
-#endif
 #ifdef DEBIAN
             case 't':
                 selected_filters++;
@@ -976,7 +954,7 @@
             selected = select_series(distro_list, series_name);
         } else {
             selected = get_distro(distro_list, date, filter_cb, select_cb);
-#ifdef UBUNTU
+#ifdef TRISQUEL
             if(selected == NULL && filter_latest) {
                 selected = get_distro(distro_list, date, filter_stable, select_latest_release);
             }
