--- src/actions-live.cpp	2017-08-29 08:36:12.000000000 -0700
+++ src/actions-live.cpp  	2019-09-10 13:06:26.971791147 -0700
@@ -82,6 +82,9 @@
                 if (!g_strcmp0 (desktop_names[i], "Unity")) {
                     result = LiveActions::UNITY7;
                     break;
+                } else if (!g_strcmp0 (desktop_names[i], "MATE")) {
+                    result = LiveActions::MATE;
+                    break;
                 }
             }
             g_strfreev (desktop_names);
@@ -144,6 +147,9 @@
     case LiveActions::UNITY7:
         execute_command("unity-control-center datetime");
         break;
+    case LiveActions::MATE:
+        execute_command("time-admin");
+        break;
     default:
         execute_command("gnome-control-center datetime");
     }
