diff --git a/services/settings/RemoteSettingsClient.sys.mjs b/services/settings/RemoteSettingsClient.sys.mjs
index 7e98e6d..7716e41 100644
--- a/services/settings/RemoteSettingsClient.sys.mjs
+++ b/services/settings/RemoteSettingsClient.sys.mjs
@@ -227,6 +227,8 @@ class AttachmentDownloader extends Downloader {
    * @see Downloader.download
    */
   async download(record, options) {
+    console.warn("Function 'download' disabled in Abrowser due privacy concerns.");
+    return null;
     try {
       // Explicitly await here to ensure we catch a network error.
       return await super.download(record, options);
diff --git a/services/settings/Utils.sys.mjs b/services/settings/Utils.sys.mjs
index 12fef6c..c52b65e 100644
--- a/services/settings/Utils.sys.mjs
+++ b/services/settings/Utils.sys.mjs
@@ -409,6 +409,8 @@ export var Utils = {
    * @param {Object} filters
    */
   async fetchLatestChanges(serverUrl, options = {}) {
+    console.warn("Function 'fetchLatestChanges' disabled in Abrowser due privacy concerns.");
+    return null;
     const { expectedTimestamp, lastEtag = "", filters = {} } = options;
 
     let url = serverUrl + Utils.CHANGES_PATH;
