diff -ru choose-mirror-2.78ubuntu7+10.0trisquel3/Makefile choose-mirror-2.111/Makefile
--- choose-mirror-2.78ubuntu7+10.0trisquel3/Makefile	2021-02-26 15:22:56.000000000 -0600
+++ choose-mirror-2.111/Makefile	2021-06-04 12:09:55.000000000 -0500
@@ -33,8 +33,8 @@
 STRIP=strip
 
 # Derivative distributions may want to change these.
-#MIRRORLISTURL=https://anonscm.debian.org/git/mirror/mirror-masterlist.git/plain/Mirrors.masterlist
-MASTERLIST=Mirrors.masterlist.trisquel
+MIRRORLISTURL=https://salsa.debian.org/mirror-team/masterlist/raw/master/Mirrors.masterlist
+MASTERLIST=Mirrors.masterlist
 
 ifdef DEBUG
 CFLAGS:=$(CFLAGS) -DDODEBUG
