--- a/setup.py	2024-08-24 23:17:50.141778711 -0400
+++ b/setup.py	2024-08-24 23:11:05.728064936 -0400
@@ -1,6 +1,7 @@
 #!/usr/bin/env python3
 
 import glob
+import re
 
 from distutils.core import setup
 from subprocess import check_output
@@ -15,6 +16,7 @@
         # PEP 440 uses '!' for the epoch separator:
         #   https://peps.python.org/pep-0440/#version-epochs
         version = value.strip().replace(':','!')
+        version = re.sub("([a-zA-Z])", r"+\1", version, count=1)
         break
 else:
     raise RuntimeError('No version found in debian/changelog')
