--- debian/supported-versions.orig	2016-03-29 04:53:13.000000000 -0400
+++ debian/supported-versions	2017-02-10 14:20:08.503521825 -0500
@@ -55,9 +55,12 @@
 
 default() {
     case "$DISTRO" in
+        [tT]risquel)
+            trisquel "$RELEASE"
+            ;;
         [uU]buntu)
             ubuntu "$RELEASE"
             ;;
         [dD]ebian)
             debian "$RELEASE"
             ;;
@@ -76,6 +82,36 @@
             ;;
     esac
 }
+
+trisquel() {
+    case "$1" in
+        4.0*) # Taranis
+            /bin/echo -e "8.4"
+            ;;
+        6.0*) # Toutatis
+            /bin/echo -e "9.1"
+            ;;
+        7.0*) # Belenos
+            /bin/echo -e "9.3"
+            ;;
+        8.0*) # Flidas
+            /bin/echo -e "9.5"
+            ;;
+        9.0*) # Etiona
+            /bin/echo -e "10"
+            ;;
+        10.0*) # Nabia
+            /bin/echo -e "12"
+            ;;
+        11.0*) # Aramo
+            /bin/echo -e "14"
+            ;;
+        *)
+            echo "supported_versions: WARNING: Unknown Trisquel release: $1" >&2
+            /bin/echo -e "$DEFAULT"
+            ;;
+    esac
+}
 
 ubuntu() {
     case "$1" in
