diff --git a/softwareproperties/ppa.py b/softwareproperties/ppa.py
index a979a3c9..c4e7cadc 100644
--- a/softwareproperties/ppa.py
+++ b/softwareproperties/ppa.py
@@ -32,6 +32,8 @@
 
 from urllib.parse import urlparse
 
+from softwareproperties.trisquel_info import trisquel_upstream_rel
+
 
 PPA_URI_FORMAT = 'https://ppa.launchpadcontent.net/{team}/{ppa}/ubuntu/'
 PRIVATE_PPA_URI_FORMAT = 'https://private-ppa.launchpadcontent.net/{team}/{ppa}/ubuntu/'
@@ -98,7 +98,7 @@ class PPAShortcutHandler(ShortcutHandler):
             entry = Deb822SourceEntry(None, '')
             entry.types = [self.binary_type]
             entry.uris = [uri]
-            entry.suites = [self.dist]
+            entry.suites = [trisquel_upstream_rel]
             entry.comps = comps
 
             self._set_source_entry(str(entry))
