diff --git a/po/es.po b/po/es.po
index aced39c..6c4c678 100644
--- a/po/es.po
+++ b/po/es.po_
@@ -41,10 +41,10 @@ msgstr "Borrar disco"
 
 #: .././gui/usbcreator-gtk.ui.h:6
 msgid ""
-"Installation is complete.  You may now run Trisquel on other computers by "
+"Installation is complete.  You may now run your OS on other computers by "
 "booting them with this drive inserted."
 msgstr ""
-"La instalación se ha completado. Ahora puede ejecutar Trisquel en otros "
+"La instalación se ha completado. Ahora puede ejecutar su SO en otros "
 "equipos arrancándolos con el disco insertado."
 
 #. set title of progress window (same as gtk frontend)
@@ -69,10 +69,10 @@ msgstr "Crear disco de inicio"
 
 #: .././gui/usbcreator-gtk.ui.h:11
 msgid ""
-"To try or install Trisquel from a removable disk, it needs to be set up as a "
+"To try or install your OS from a removable disk, it needs to be set up as a "
 "startup disk."
 msgstr ""
-"Para probar o instalar Trisquel desde un disco extraíble, es necesario "
+"Para probar o instalar su SO desde un disco extraíble, es necesario "
 "establecerlo como disco de arranque."
 
 #: .././gui/usbcreator-gtk.ui.h:12
@@ -198,10 +198,10 @@ msgstr "Finalizando..."
 #: .././usbcreator/frontends/kde/frontend.py:487
 msgid ""
 "The installation is complete.  You may now reboot your computer with this "
-"device inserted to try or install Trisquel."
+"device inserted to try or install your OS."
 msgstr ""
 "La instalación se ha completado. Debe reiniciar su equipo con este "
-"dispositivo insertado para probar o instalar Trisquel."
+"dispositivo insertado para probar o instalar su SO."
 
 #: .././usbcreator/frontends/kde/frontend.py:499
 msgid "Retry?"
