diff --git a/po/fr.po b/po/fr.po_
index 9e7a9df..dfe4c0f 100644
--- a/po/fr.po
+++ b/po/fr.po_
@@ -42,11 +42,12 @@ msgstr "Effacer le disque"
 
 #: .././gui/usbcreator-gtk.ui.h:6
 msgid ""
-"Installation is complete.  You may now run Trisquel on other computers by "
+"Installation is complete.  You may now run your OS on other computers by "
 "booting them with this drive inserted."
 msgstr ""
-"L'installation est terminée. Vous pouvez maintenant utiliser Trisquel sur "
-"d'autres ordinateurs en les faisant démarrer sur cette clé USB."
+"L’installation est terminée.  Vous pouvez maintenant utiliser "
+"le système d’exploitation sur d’autres ordinateurs "
+"en y insérant ce disque et en démarrant dessus."
 
 #. set title of progress window (same as gtk frontend)
 #: .././gui/usbcreator-gtk.ui.h:7
@@ -70,11 +71,11 @@ msgstr "Créer un disque de démarrage"
 
 #: .././gui/usbcreator-gtk.ui.h:11
 msgid ""
-"To try or install Trisquel from a removable disk, it needs to be set up as a "
+"To try or install your OS from a removable disk, it needs to be set up as a "
 "startup disk."
 msgstr ""
-"Pour essayer ou installer Trisquel depuis un disque amovible, il doit être "
-"défini comme disque de démarrage."
+"Pour essayer ou installer le système d’exploitation depuis un disque amovible, "
+"il doit y être configuré comme disque de démarrage."
 
 #: .././gui/usbcreator-gtk.ui.h:12
 #, fuzzy
@@ -201,10 +202,11 @@ msgstr "Finalisation..."
 #: .././usbcreator/frontends/kde/frontend.py:487
 msgid ""
 "The installation is complete.  You may now reboot your computer with this "
-"device inserted to try or install Trisquel."
+"device inserted to try or install your OS."
 msgstr ""
-"L'installation est terminée. Vous pouvez maintenant redémarrer votre "
-"ordinateur avec le périphérique inséré pour essayer ou installer Trisquel."
+"L’installation est terminée.  Vous pouvez maintenant redémarrer "
+"votre ordinateur avec ce disque inséré pour essayer ou installer "
+"le système d’exploitation."
 
 #: .././usbcreator/frontends/kde/frontend.py:499
 msgid "Retry?"
