--- a/validate-csv-data
+++ b/validate-csv-data
@@ -39,6 +39,15 @@


 _COLUMNS = {
+    "trisquel": (
+        "version",
+        "codename",
+        "series",
+        "created",
+        "release",
+        "eol",
+        "upstream",
+    ),
     "debian": (
         "version",
         "codename",
@@ -92,6 +101,7 @@
     ("eol-esm", "eol-legacy"),
 )
 _STRINGS = {
+    "trisquel": ("version", "codename", "series"),
     "debian": ("codename", "series"),
     "devuan": ("codename", "series"),
     "elxr": ("codename", "series"),
