--- a/validate-csv-data	2021-10-15 08:01:00.000000000 -0500
+++ b/validate-csv-data	2022-04-06 12:27:29.004706669 -0500
@@ -27,6 +27,13 @@
 
 
 _COLUMNS = {
+    "trisquel": (
+        "version",
+        "codename",
+        "series",
+        "created",
+        "release",
+        "eol"),
     "debian": (
         "version",
         "codename",
@@ -58,6 +65,7 @@
     ("eol-lts", "eol-elts"),
 )
 _STRINGS = {
+    "trisquel": ("version", "codename", "series"),
     "debian": ("codename", "series"),
     "ubuntu": ("version", "codename", "series"),
 }
