diff -Naur mate-control-center-1.20.2-c/capplets/default-applications/mate-da-capplet.h mate-control-center-1.20.2-f461/capplets/default-applications/mate-da-capplet.h
--- mate-control-center-1.20.2-c/capplets/default-applications/mate-da-capplet.h	2018-03-27 06:15:26.000000000 -0600
+++ mate-control-center-1.20.2-f461/capplets/default-applications/mate-da-capplet.h	2020-01-26 17:19:36.795605940 -0600
@@ -29,7 +29,8 @@
 
 #define VISUAL_SCHEMA         "org.mate.applications-at-visual"
 #define VISUAL_KEY            "exec"
-#define VISUAL_STARTUP_KEY    "startup"
+#define VISUAL_STARTUP_SCHEMA "org.gnome.desktop.a11y.applications"
+#define VISUAL_STARTUP_KEY    "screen-reader-enabled"
 
 #define MOBILITY_SCHEMA       "org.mate.applications-at-mobility"
 #define MOBILITY_KEY          "exec"
@@ -80,6 +81,7 @@
 	/* Settings objects */
 	GSettings* terminal_settings;
 	GSettings* visual_settings;
+	GSettings* visual_startup_settings;
 	GSettings* mobility_settings;
 } MateDACapplet;
 
