diff --git a/browser/components/BrowserGlue.sys.mjs b/browser/components/BrowserGlue.sys.mjs
index 8fa6f7a..a34ab8b 100644
--- a/browser/components/BrowserGlue.sys.mjs
+++ b/browser/components/BrowserGlue.sys.mjs
@@ -1860,6 +1860,19 @@ BrowserGlue.prototype = {
       }
     });
 
+    // Trisquel hack: if no user settings for tracking, default to strict instead of standard.
+    if (!
+      (Services.prefs.prefHasUserValue("browser.contentblocking.category")
+      || Services.prefs.prefHasUserValue("network.cookie.cookieBehavior")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.socialtracking.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.cryptomining.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.fingerprinting.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.pbmode.enabled"))
+    ) {
+      Services.prefs.setStringPref("browser.contentblocking.category", "strict"); this._updateCBCategory;
+    }
+
     this._maybeOfferProfileReset();
 
     this._checkForOldBuildUpdates();
