diff -ru firefox-110.0+build1/browser/components/BrowserGlue.sys.mjs firefox-110.0+build1/browser/components/BrowserGlue.sys.mjs_fix
--- firefox-110.0+build1/browser/components/BrowserGlue.sys.mjs	2023-02-07 01:52:32.000000000 -0600
+++ firefox-110.0+build1/browser/components/BrowserGlue.sys.mjs_fix	2023-02-07 14:52:59.465762604 -0600
@@ -1637,6 +1637,19 @@
       }
     });
 
+    // Trisquel hack: if no user settings for tracking, default to strict instead of standard.
+    if (!
+      (Services.prefs.prefHasUserValue("browser.contentblocking.category")
+      || Services.prefs.prefHasUserValue("network.cookie.cookieBehavior")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.socialtracking.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.cryptomining.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.fingerprinting.enabled")
+      || Services.prefs.prefHasUserValue("privacy.trackingprotection.pbmode.enabled"))
+    ) {
+      Services.prefs.setStringPref("browser.contentblocking.category", "strict"); this._updateCBCategory;
+    }
+
     // Offer to reset a user's profile if it hasn't been used for 60 days.
     const OFFER_PROFILE_RESET_INTERVAL_MS = 60 * 24 * 60 * 60 * 1000;
     let lastUse = Services.appinfo.replacedLockTime;
