From dd0f9279eb831c49ac73b15f565bc9abcf9005ed Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Sun, 31 Jan 2021 20:52:34 +0100
Subject: [PATCH] i18n: Do not collect the translation for Icon (data folder)

---
 data/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index e3e0eb9f..b7056353 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -14,7 +14,7 @@ desktop_in_files = mate-screensaver-preferences.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
 $(desktop_DATA): $(desktop_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 directorydir = $(datadir)/desktop-directories
 directory_in_files =                            \
@@ -23,7 +23,7 @@ directory_in_files =                            \
 directory_DATA = $(directory_in_files:.directory.desktop.in=.directory)
 
 $(directory_DATA): $(directory_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --template $< -d $(top_srcdir)/po -o $@
 
 menudir = $(sysconfdir)/xdg/menus
 menu_DATA =					\
