From 51e6322db5717c7baa9a517a19af163fbeed56a5 Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Fri, 15 Jan 2021 18:18:39 +0100
Subject: [PATCH] i18n: Do not collect the translation for Icon

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index a1c65ed9..f26612be 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -57,7 +57,7 @@ desktopdir = $(sysconfdir)/xdg/autostart
 desktop_in_files = mate-screensaver.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 $(desktop_DATA): $(desktop_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 mate_screensaver_command_SOURCES = 	\
 	mate-screensaver-command.c	\
