From 91962719d06ce16d8bc3523872b83fae4d151e10 Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Mon, 2 Mar 2020 08:53:51 +0100
Subject: [PATCH] mate-calc.desktop: Do not collect the translation for Icon

---
 data/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index 80db3448..f12b6dac 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -11,7 +11,7 @@ Utilitiesdir = $(datadir)/applications
 Utilities_in_files = mate-calc.desktop.in
 Utilities_DATA = $(Utilities_in_files:.desktop.in=.desktop)
 $(Utilities_DATA): $(Utilities_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=GenericName --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 man1_MANS = mate-calc.1 mate-calc-cmd.1
 
