--- a/UpdateManager/Core/utils.py	2015-10-21 04:05:26.461825091 +0200
+++ b/UpdateManager/Core/utils.py	2015-10-21 04:03:30.461825105 +0200
@@ -426,7 +426,7 @@
     # From spec, first if ubuntu-desktop is installed, use that.
     # Second, grab first installed one from DistUpgrade.cfg.
     # Lastly, fallback to ubuntu-desktop again.
-    meta_pkgs = ['ubuntu-desktop']
+    meta_pkgs = ['trisquel','trisquel-mini','trisquel-sugar', 'triskel', 'trisquel-gnome']
 
     try:
         meta_pkgs.extend(sorted(_load_meta_pkg_list()))
@@ -439,13 +439,13 @@
         cache_pkg = cache[meta_pkg] if meta_pkg in cache else None
         if cache_pkg and cache_pkg.is_installed:
             return meta_pkg
-    return 'ubuntu-desktop'
+    return 'trisquel'
 
 
 def get_ubuntu_flavor_name(cache=None):
     """ try to guess the flavor name based on the running desktop """
     pkg = get_ubuntu_flavor_package(cache=cache)
-    lookup = {'ubuntustudio-desktop': 'Ubuntu Studio'}
+    lookup = {'trisquel-mini': 'Trisquel Mini', 'trisquel-sugar': 'Trisquel Sugar' }
     if pkg in lookup:
         return lookup[pkg]
     elif pkg.endswith('-desktop'):
