Index: rekonq-0.6.80/src/rekonq.kcfg
===================================================================
--- rekonq-0.6.80.orig/src/rekonq.kcfg	2011-02-08 17:12:18.000000000 +0000
+++ rekonq-0.6.80/src/rekonq.kcfg	2011-02-09 12:11:25.094984734 +0000
@@ -59,7 +59,7 @@
         <default>0</default>
     </entry>
     <entry name="homePage" type="String">
-        <default>http://www.kde.org/</default>
+        <default>http://trisquel.info/</default>
     </entry>
     <entry name="useNewTabPage" type="Bool">
         <default>false</default>
