Index: rekonq-0.6.95+git20110331/src/data/rekonq.desktop
===================================================================
--- rekonq-0.6.95+git20110331.orig/src/data/rekonq.desktop	2011-03-31 17:08:13.913727631 +0100
+++ rekonq-0.6.95+git20110331/src/data/rekonq.desktop	2011-03-31 17:08:18.333727630 +0100
@@ -83,4 +83,4 @@
 MimeType=text/html;application/xhtml+xml;application/xml;
 X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=org.kde.rekonq
-
+InitialPreference=10
