From 74f37ee5727f50f4175dbba9744391e3245e7982 Mon Sep 17 00:00:00 2001
From: "I.Rezkov" <fagothmail@gmail.com>
Date: Fri, 31 Oct 2025 21:06:08 +0300
Subject: [PATCH] fix downloading screenshots

---
 common/rpackage.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/rpackage.cc b/common/rpackage.cc
index 834b199d..3b40fbc2 100644
--- a/common/rpackage.cc
+++ b/common/rpackage.cc
@@ -949,9 +949,9 @@ string RPackage::getScreenshotFile(pkgAcquire *fetcher, bool thumb)
       verstr=string(verstr, verstr.find(':')+1);
    char uri[512];
    if(thumb)
-      snprintf(uri,512,"http://screenshots.debian.net/thumbnail/%s", name());
+      snprintf(uri,512,"https://screenshots.debian.net/thumbnail/%s", name());
    else
-      snprintf(uri,512,"http://screenshots.debian.net/screenshot/%s", name());
+      snprintf(uri,512,"https://screenshots.debian.net/screenshot/%s", name());
 
    //cerr << "uri is: " << uri << endl;
 
