From d245a2f56302b7a26528194f30c788f5ab80e38f Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Sun, 31 Jan 2021 22:28:37 +0100
Subject: [PATCH] i18n: Do not collect the translation for Icon

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 95f16ecf..d7e22e71 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,7 +41,7 @@ Applications_in_files = mate-system-monitor.desktop.in
 Applications_DATA = $(Applications_in_files:.desktop.in=.desktop)
 
 $(Applications_DATA): $(Applications_in_files)
-	$(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+	$(AM_V_GEN) $(MSGFMT) --desktop --keyword= --keyword=Name --keyword=Comment --keyword=Keywords --template $< -d $(top_srcdir)/po -o $@
 
 distuninstallcheck_listfiles = find . -type f -print
 
