=== modified file 'build/Makefile'
--- build/Makefile	2010-11-05 16:34:29 +0000
+++ build/Makefile	2011-03-01 11:42:39 +0000
@@ -294,6 +294,7 @@
 	rm -rf $(TREE)
 	# Set up the basic files [u]dpkg needs.
 	mkdir -p $(DPKGDIR)/info
+	ln -nsf . $(DPKGDIR)/info/$(shell dpkg --print-architecture)
 	touch $(DPKGDIR)/status
 	# Create a tmp tree
 	mkdir -p $(TREE)/tmp
@@ -457,6 +458,7 @@
 
 	mkdir -p $(EXTRAUDEBSDIR)
 	mkdir -p $(EXTRAUDEBSDPKGDIR)/info $(EXTRAUDEBSDPKGDIR)/updates
+	ln -nsf . $(EXTRAUDEBSDPKGDIR)/info/$(shell dpkg --print-architecture)
 	touch $(EXTRAUDEBSDPKGDIR)/status $(EXTRAUDEBSDPKGDIR)/available
 
 ifdef EXTRADRIVERS

#=== modified file 'debian/changelog'
#--- debian/changelog	2011-03-01 10:36:50 +0000
#+++ debian/changelog	2011-03-01 11:42:39 +0000
#@@ -1,3 +1,10 @@
#+debian-installer (20101020ubuntu21) UNRELEASED; urgency=low
#+
#+  * Install /var/lib/dpkg/info/$(dpkg --print-architecture) -> . symlink so
#+    that images build correctly in the presence of multiarch dpkg.
#+
#+ -- Colin Watson <cjwatson@ubuntu.com>  Tue, 01 Mar 2011 11:41:38 +0000
#+
# debian-installer (20101020ubuntu20) natty; urgency=low
# 
#   * Stop building the armel dove subarchitecture; it no longer has a kernel
#
