--- data/gtkbuilder/main.ui	2021-11-26 16:28:28.539391504 -0600
+++ data/gtkbuilder/main.ui_etiona	2021-11-26 16:30:29.779857087 -0600
@@ -553,18 +553,6 @@
                   </packing>
                 </child>
                 <child>
-                  <object class="GtkLabel">
-                    <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="label" translatable="yes">Snap package updates are checked routinely and installed automatically.</property>
-                    <property name="xalign">0</property>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">False</property>
-                  </packing>
-                </child>
-                <child>
                   <object class="GtkAlignment" id="alignment2">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
