diff --git a/kernel/tests/armhf/efikamx.test b/kernel/tests/armhf/efikamx.test
deleted file mode 100644
index 3fa4eeef..00000000
--- a/kernel/tests/armhf/efikamx.test
+++ /dev/null
@@ -1,19 +0,0 @@
-subarch mx5
-cpuinfo efikamx.cpuinfo
-majors 5.7
-flavour mx5 armmp
-kernel-5.7 linux-image-mx5 linux-image-armmp
-usable \
-  linux-image-mx5 \
-  linux-image-5.7.0-1-mx5 \
-  linux-image-armmp \
-  linux-image-5.7.0-1-armmp
-unusable \
-  linux-image-armmp-lpae \
-  linux-image-netwinder \
-  linux-image-orion5x \
-  linux-image-rpc \
-  linux-image-s3c2410 \
-  linux-image-kirkwood \
-  linux-image-vexpress \
-  linux-image-5.7.0-1-armmp-dbg
diff --git a/kernel/tests/armhf/highbank.test b/kernel/tests/armhf/highbank.test
index 1f68d466..37a42821 100644
--- a/kernel/tests/armhf/highbank.test
+++ b/kernel/tests/armhf/highbank.test
@@ -1,14 +1,14 @@
 subarch generic
 cpuinfo highbank.cpuinfo
 majors 5.7
-flavour armmp
+flavour generic
 kernel-5.7 \
-  linux-image-armmp
+  linux-image-generic
 usable \
-  linux-image-armmp \
-  linux-image-5.7.0-1-armmp
+  linux-image-generic \
+  linux-image-5.7.0-1-generic
 unusable \
-  linux-image-armmp-lpae \
+  linux-image-generic-lpae \
   linux-image-netwinder \
   linux-image-orion5x \
   linux-image-rpc \
@@ -16,4 +16,4 @@ unusable \
   linux-image-kirkwood \
   linux-image-mx5 \
   linux-image-vexpress \
-  linux-image-5.7.0-1-armmp-dbg
+  linux-image-5.7.0-1-generic-dbg
diff --git a/kernel/tests/armhf/midway-no-lpae.test b/kernel/tests/armhf/midway-no-lpae.test
index 04db5356..056ff08f 100644
--- a/kernel/tests/armhf/midway-no-lpae.test
+++ b/kernel/tests/armhf/midway-no-lpae.test
@@ -1,14 +1,14 @@
 subarch generic
 cpuinfo midway-no-lpae.cpuinfo
 majors 5.7
-flavour armmp
+flavour generic
 kernel-5.7 \
-  linux-image-armmp
+  linux-image-generic
 usable \
-  linux-image-armmp \
-  linux-image-5.7.0-1-armmp
+  linux-image-generic \
+  linux-image-5.7.0-1-generic
 unusable \
-  linux-image-armmp-lpae \
+  linux-image-generic-lpae \
   linux-image-netwinder \
   linux-image-orion5x \
   linux-image-rpc \
@@ -16,4 +16,4 @@ unusable \
   linux-image-kirkwood \
   linux-image-mx5 \
   linux-image-vexpress \
-  linux-image-5.7.0-1-armmp-dbg
+  linux-image-5.7.0-1-generic-dbg
diff --git a/kernel/tests/armhf/midway.test b/kernel/tests/armhf/midway.test
index 6a789a1b..a5213a4a 100644
--- a/kernel/tests/armhf/midway.test
+++ b/kernel/tests/armhf/midway.test
@@ -1,15 +1,15 @@
 subarch generic
 cpuinfo midway.cpuinfo
 majors 5.7
-flavour armmp-lpae armmp
+flavour generic-lpae generic
 kernel-5.7 \
-  linux-image-armmp-lpae \
-  linux-image-armmp
+  linux-image-generic-lpae \
+  linux-image-generic
 usable \
-  linux-image-armmp \
-  linux-image-5.7.0-1-armmp \
-  linux-image-armmp-lpae \
-  linux-image-5.7.0-1-armmp-lpae
+  linux-image-generic \
+  linux-image-5.7.0-1-generic \
+  linux-image-generic-lpae \
+  linux-image-5.7.0-1-generic-lpae
 unusable \
   linux-image-netwinder \
   linux-image-orion5x \
@@ -18,4 +18,4 @@ unusable \
   linux-image-kirkwood \
   linux-image-mx5 \
   linux-image-vexpress \
-  linux-image-5.7.0-1-armmp-dbg
+  linux-image-5.7.0-1-generic-dbg
diff --git a/kernel/tests/armhf/vexpress.test b/kernel/tests/armhf/vexpress.test
deleted file mode 100644
index 19d38dd4..00000000
--- a/kernel/tests/armhf/vexpress.test
+++ /dev/null
@@ -1,19 +0,0 @@
-subarch vexpress
-cpuinfo vexpress.cpuinfo
-majors 5.7
-flavour vexpress armmp
-kernel-5.7 linux-image-vexpress linux-image-armmp
-usable \
-  linux-image-vexpress \
-  linux-image-3.2.0-4-vexpress \
-  linux-image-armmp \
-  linux-image-5.7.0-1-armmp
-unusable \
-  linux-image-armmp-lpae \
-  linux-image-netwinder \
-  linux-image-orion5x \
-  linux-image-rpc \
-  linux-image-s3c2410 \
-  linux-image-kirkwood \
-  linux-image-mx5 \
-  linux-image-5.7.0-1-armmp-dbg
