diff --git a/test/surface_test.py b/test/surface_test.py
index 4323503..1c161ad 100644
--- a/test/surface_test.py
+++ b/test/surface_test.py
#@@ -1481,7 +1481,7 @@ class GeneralSurfaceTests(AssertRaisesRegexMixin, unittest.TestCase):
#         self.assertEqual(results, results_expected)
# 
#     @unittest.skipIf(
#-        "arm" in platform.machine() or "aarch64" in platform.machine() or "mips" in platform.machine() or "ppc64le" in platform.machine() or "s390x" in platform.machine(),
#+        "arm" in platform.machine() or "aarch64" in platform.machine() or "mips" in platform.machine() or "ppc64le" in platform.machine() or "s390x" in platform.machine() or "riscv64" in platform.machine(),
#         "sdl2 blitter produces different results on arm",
#     )
#     def test_src_alpha_sdl2_blitter(self):
@@ -2013,7 +2013,7 @@ class GeneralSurfaceTests(AssertRaisesRegexMixin, unittest.TestCase):
             mask8 = (224, 28, 3, 0)
             mask15 = (31744, 992, 31, 0)
             mask16 = (63488, 2016, 31, 0)
-            mask24 = (4278190080, 16711680, 65280, 0)
+            mask24 = (16711680, 65280, 255, 0)
             mask32 = (4278190080, 16711680, 65280, 255)
 
             # Surfaces with standard depths and masks
