diff --git a/debian/rules b/debian/rules
index 06de6171..b930f423 100755
--- a/debian/rules
+++ b/debian/rules
@@ -824,7 +824,8 @@ endif
 # for t64
 BUILD_DEPS += , dpkg-dev (>= 1.22.5) [$(filter-out i386,$(OOO_32BIT_ARCHS))]
 
-ifeq "$(DEB_VENDOR)" "Ubuntu"
+ifeq ("$(DEB_VENDOR)",$(filter "$(DEB_VENDOR)","Ubuntu" "Trisquel"))
+  ifeq "$(DEB_VENDOR)" "Ubuntu"
   # Set up Google API keys, see http://www.chromium.org/developers/how-tos/api-keys .
   # Note: these are for Ubuntu use ONLY. For your own distribution,
   # please get your own set of keys.
@@ -835,7 +835,7 @@
   GOOGLEAPI_CLIENTID_UBUNTU := 424119844901-gee57209rkbo1rgula4i0arilvgv3lsf.apps.googleusercontent.com
   GOOGLEAPI_CLIENTSECRET_UBUNTU := 3h1DknIrVsq2wEhIuADVxQ3E
   CONFIGURE_FLAGS += --with-gdrive-client-id=$(GOOGLEAPI_CLIENTID_UBUNTU) --with-gdrive-client-secret=$(GOOGLEAPI_CLIENTSECRET_UBUNTU)
-
+  endif
   # lp#1034560: MIR troubles make it easier to use internal jfreereport
   SYSTEM_STUFF := $(filter-out jfreereport npapi-headers, $(SYSTEM_STUFF))
   SYSTEM_STUFF := $(filter-out libzmf libstaroffice, $(SYSTEM_STUFF))
