--- debian/rules	2021-05-07 19:08:17.495168344 +0200
+++ debian/rules	2021-05-07 19:27:02.173815086 +0200
@@ -597,17 +597,19 @@
   BULLSEYE_BACKPORT=y
 endif

-ifeq "$(DEB_VENDOR)" "Ubuntu"
+ifeq ("$(DEB_VENDOR)",$(filter "$(DEB_VENDOR)","Ubuntu" "Trisquel"))
+  ifeq "$(DEB_VENDOR)" "Ubuntu"
   # Set up Google API keys, see http://www.chromium.org/developers/how-tos/api-keys .
   # Note: these are for Ubuntu use ONLY. For your own distribution,
   # please get your own set of keys.
   # Permission to add API keys, from Paweł Hajdan, To chad.miller@canonical.com
   # msgid: CAADNaOFSFoch68NM1SGpCTRXqmspyKQgUPUtsF7SGRsRXiHZcg@mail.gmail.com
   # reused from chromium-browser 48.0.2564.82-0ubuntu1.1222
   GOOGLEAPI_APIKEY_UBUNTU := AIzaSyDDiKg-iNf3zW2j16KDsnSDnJJAgh4TnCc
   GOOGLEAPI_CLIENTID_UBUNTU := 424119844901-gee57209rkbo1rgula4i0arilvgv3lsf.apps.googleusercontent.com
   GOOGLEAPI_CLIENTSECRET_UBUNTU := 3h1DknIrVsq2wEhIuADVxQ3E
   CONFIGURE_FLAGS += --with-gdrive-client-id=$(GOOGLEAPI_CLIENTID_UBUNTU) --with-gdrive-client-secret=$(GOOGLEAPI_CLIENTSECRET_UBUNTU)
+  endif

   # lp#1034560: MIR troubles make it easier to use internal jfreereport
   SYSTEM_STUFF := $(filter-out jfreereport npapi-headers, $(SYSTEM_STUFF))
@@ -632,7 +634,7 @@
   # lto-wrapper: fatal error: write: No space left on device
   export DEB_BUILD_MAINT_OPTIONS = optimize=-lto

-  ifeq "$(DEB_DISTRIBUTION)" "bionic"
+  ifeq ("$(DEB_DISTRIBUTION)",$(filter "$(DEB_DISTRIBUTION)","bionic" "etiona" "etiona-backports"))
     SYSTEM_STUFF := $(filter-out cppunit xmlsec, $(SYSTEM_STUFF))

     SYSTEM_STUFF := $(filter-out mdds orcus libcmis, $(SYSTEM_STUFF))
