diff -Nru debian/changelog debian/changelog
--- debian/changelog	2021-11-04 09:22:24.000000000 -0600
+++ debian/changelog	2021-12-31 21:58:35.247237140 -0600
@@ -51,6 +51,19 @@
 
  -- Rico Tzschichholz <ricotz@ubuntu.com>  Fri, 07 May 2021 17:03:32 +0200
 
+  * debian/control.in, debian/libreoffice-common.{maintscript,postinst.in}:
+    apply patch from Adreas Beckmann to fix upgrade buster->bullseye
+    - libreoffice-core: Copy some Conflicts from libreoffice-common for smoother
+      upgrades from buster. Dpkg will otherwise ignore Conflicts that are
+      encountered later against a package that is already deconfigured.
+    - libreoffice-common: Do not use dir_to_symlink for
+      /usr/lib/libreoffice/share/registry, the Breaks/Conflicts cascade does not
+      work reliable here to ensure all packages previously shipping files there
+      are either removed or upgraded first, but not just deconfigured. Fix up
+      the symlink in postinst instead.  (Closes: #985297)
+
+ -- Rene Engelhard <rene@debian.org>  Sat, 01 May 2021 13:14:17 +0200
+
 libreoffice (1:7.1.2~rc2-0ubuntu2) hirsute; urgency=medium
 
   [ Heather Ellsworth ]
diff -Nru debian/control debian/control
--- debian/control	2021-11-04 09:22:24.000000000 -0600
+++ debian/control	2021-12-31 22:00:27.264676545 -0600
@@ -425,6 +425,15 @@
            libreoffice-core-nogui,
            libreoffice-filter-binfilter,
            libreoffice-mysql-connector (<< 1:6.2.0~)
+# for bullseye, copied from libreoffice-common, see #985297
+ ,
+ libreoffice-base (<< 1:7.0.0~alpha~),
+ libreoffice-calc (<< 1:7.0.0~alpha~),
+ libreoffice-draw (<< 1:7.0.0~alpha~),
+ libreoffice-impress (<< 1:7.0.0~alpha~),
+ libreoffice-math (<< 1:7.0.0~alpha~),
+ libreoffice-report-builder (<< 1:7.0.0~alpha~),
+ libreoffice-writer (<< 1:7.0.0~alpha~),
 Replaces: libreoffice-avmedia-backend-gstreamer,
           libreoffice-common (<< 1:6.3.0~rc1~),
           libreoffice-core-nogui,
diff -Nru debian/control.in debian/control.in
--- debian/control.in	2021-11-04 09:22:24.000000000 -0600
+++ debian/control.in	2021-12-31 22:01:21.745375084 -0600
@@ -238,6 +238,15 @@
         libreoffice-avmedia-backend-gstreamer (<< ${binary:Version})
 Suggests: ${gstreamer-plugins-suggests}
 Conflicts: libreoffice-filter-binfilter, libreoffice-avmedia-backend-vlc, libreoffice-mysql-connector (<< 1:6.2.0~), libreoffice-core-nogui
+# for bullseye, copied from libreoffice-common, see #985297
+ ,
+ libreoffice-base (<< 1:7.0.0~alpha~),
+ libreoffice-calc (<< 1:7.0.0~alpha~),
+ libreoffice-draw (<< 1:7.0.0~alpha~),
+ libreoffice-impress (<< 1:7.0.0~alpha~),
+ libreoffice-math (<< 1:7.0.0~alpha~),
+ libreoffice-report-builder (<< 1:7.0.0~alpha~),
+ libreoffice-writer (<< 1:7.0.0~alpha~),
 Replaces: libreoffice-pdfimport (<< 1:5.4~), libreoffice-common (<< 1:6.3.0~rc1~), libreoffice-avmedia-backend-gstreamer, libreoffice-core-nogui
 Description: office productivity suite -- arch-dependent files
  LibreOffice is a full-featured office productivity suite that provides
diff -Nru debian/libreoffice-common.maintscript debian/libreoffice-common.maintscript
--- debian/libreoffice-common.maintscript	2021-11-04 09:22:24.000000000 -0600
+++ debian/libreoffice-common.maintscript	2021-12-31 22:03:27.590985432 -0600
@@ -2,4 +2,10 @@
 mv_conffile /etc/apparmor.d/usr.lib.libreofficeprogram.senddoc /etc/apparmor.d/usr.lib.libreoffice.program.senddoc 1:5.4.4-0ubuntu1
 mv_conffile /etc/apparmor.d/usr.lib.libreofficeprogram.soffice.bin /etc/apparmor.d/usr.lib.libreoffice.program.soffice.bin 1:5.4.4-0ubuntu1
 mv_conffile /etc/apparmor.d/usr.lib.libreofficeprogram.xpdfimport /etc/apparmor.d/usr.lib.libreoffice.program.xpdfimport 1:5.4.4-0ubuntu1
-dir_to_symlink /usr/lib/libreoffice/share/registry /etc/libreoffice/registry 1:7.0.2-0ubuntu1
+
+# do this manually since dpkg-maintscript-helper dir_to_symlink
+# does not work reliably in this case because we cannot ensure that all
+# conflicting packages previously shipping files in
+# /usr/lib/libreoffice/share/registry are either upgraded or removed
+# but not just deconfigured, see #985297
+#dir_to_symlink /usr/lib/libreoffice/share/registry /etc/libreoffice/registry 1:7.0.2~rc1-1
diff -Nru debian/libreoffice-common.postinst.in debian/libreoffice-common.postinst.in
--- debian/libreoffice-common.postinst.in	2021-11-04 09:22:24.000000000 -0600
+++ debian/libreoffice-common.postinst.in	2021-12-31 22:04:25.639726958 -0600
@@ -27,3 +27,17 @@
 
 #DEBHELPER#
 
+# do this manually since dpkg-maintscript-helper dir_to_symlink
+# does not work reliably in this case because we cannot ensure that all
+# conflicting packages previously shipping files in
+# /usr/lib/libreoffice/share/registry are either upgraded or removed
+# but not just deconfigured, see #985297
+if [ "$1" = "configure" ] && dpkg --compare-versions "$2" lt-nl "1:7.0.4-4~" ; then
+       if [ ! -L /usr/lib/libreoffice/share/registry ]; then
+               if [ -d /usr/lib/libreoffice/share/registry ]; then
+                       # this will fail if the directory is not yet empty
+                       rmdir -v /usr/lib/libreoffice/share/registry
+               fi
+               ln -sfvT /etc/libreoffice/registry /usr/lib/libreoffice/share/registry
+       fi
+fi
