From e99664c67254fee3df1b53caa2e2a67b0bb86576 Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Sun, 19 Jan 2014 18:27:28 +0100
Subject: set G_APPLICATION_HANDLES_COMMAND_LINE for the application

[bug #728341]
---
 gthumb/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gthumb/main.c b/gthumb/main.c
index f3f887a..3986c7d 100644
--- a/gthumb/main.c
+++ b/gthumb/main.c
@@ -716,7 +716,7 @@ gth_application_new (void)
         return g_object_new (gth_application_get_type (),
                              "application-id", "org.gnome.Gthumb",
                              "register-session", TRUE, /* required to call gtk_application_inhibit */
-                             "flags", 0,
+                             "flags", G_APPLICATION_HANDLES_COMMAND_LINE,
                              NULL);
 }
 
