--- a/marco-compton     2016-04-11 09:19:13.000000000 -0400
+++ b/marco-compton     2017-04-19 23:04:50.406817419 -0400
@@ -40,6 +40,7 @@
         --mark-wmwin-focused \
         -r 10 -o 0.225 -l -12 -t -12 \
         -c -C -G \
+        --unredir-if-possible-exclude "class_g = 'Mate-screensaver'" \
         --shadow-exclude "! name~=''" \
         --shadow-exclude "name = 'Notification'" \
         --shadow-exclude "name = 'Plank'" \
--- a/metacity-compton  2016-04-11 09:19:13.000000000 -0400
+++ b/metacity-compton  2017-04-19 23:04:35.926817421 -0400
@@ -40,6 +40,7 @@
         --mark-wmwin-focused \
         -r 7 -o 0.09 -l -8 -t -8 \
         -c -C -G \
+        --unredir-if-possible-exclude "class_g = 'Mate-screensaver'" \
         --shadow-exclude "! name~=''" \
         --shadow-exclude "name = 'Notification'" \
         --shadow-exclude "name = 'Plank'" \
